File: ios/PrivateLine/Message.swift
/*
 * Message.swift - Simple models for chat messages and groups.
 * Decodes JSON returned by the backend API.
 */
import Foundation

/// Basic model representing a chat message returned by the API.
struct Message: Identifiable, Decodable {
    /// Unique identifier for the message.
    let id: Int
    /// Decrypted text content.
    let content: String
    /// Optional id referencing an uploaded attachment
    let file_id: Int?
    /// Whether the message has been read by the recipient
    let read: Bool?
    /// ISO8601 timestamp when the message expires, nil means it is permanent
    let expires_at: Date?
}

struct Group: Identifiable, Decodable {
    /// Unique identifier for the group.
    let id: Int
    /// Human readable group name.
    let name: String
}

File: android/app/src/main/java/com/example/privateline/Message.kt
/**
 * Message.kt - Data models for chat objects.
 *
 * Provides simple representations matching the backend JSON structure so
 * the Android client can decode responses and persist message state such
 * as read receipts.
 */
package com.example.privateline

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * Model representing a single chat message.
 *
 * @property id Unique identifier for the message.
 * @property content Decrypted text body of the message.
 * @property file_id Optional attachment identifier returned when a file is uploaded.
 * @property read Whether the message has been read by the recipient.
 * @property expires_at Optional timestamp when the message should expire locally.
 */
data class Message(
    val id: Int,
    val content: String,
    @SerializedName("file_id") val file_id: Int?,
    val read: Boolean?,
    @SerializedName("expires_at") val expires_at: Date?
)
